wiki/Introduction-to-Creating-a-Job.md

```diff
@@ -0,0 +1,1818 @@
+
+```bash
+#!/bin/bash
+
+set -eou pipefail
+
+INPUT_DIR="$1"
+OUTPUT_FILENAME="$2"
+START_FRAME="$3"
+
+ffmpeg -y -r 10 -start_number "$START_FRAME" -i "$INPUT_DIR"/frame-%03d.png -pix_fmt yuv420p \
+    -vf "scale=in_color_matrix=bt709:out_color_matrix=bt709" \
+    -frames:v 300 -c:v libx264 -preset fast \
+    -color_range tv -colorspace bt709 -color_primaries bt709 -color_trc iec61966-2-1 \
+    -movflags faststart "$OUTPUT_FILENAME"
+```
+
+and then test it again:
+
+```bash
+% ./encode.sh output_frames animation.mp4 1                                           
+ffmpeg version 6.1.1-tessus  https://evermeet.cx/ffmpeg/  Copyright (c) 2000-2023 the FFmpeg developers
+...
+```
+
+## 2.2. Creating a Job Template
+
+Once you have a set of shell scripts that accomplish the goals of your Job, the next step is to
+create a Job Template from those shell scripts so that you can run the Job on your distributed
+computing infrastructure. The Job Template describes the shape of the Job, its runtime environment,
+and the processes that will run.
+
+### 2.2.1. Embed the scripts into a Job Template
+
+Starting with a skeleton of the minimal required properties of a Job Template:
+
+```yaml
+specificationVersion: jobtemplate-2023-09
+name: 
+steps:
+  - name: 
+    script:
+      actions:
+        onRun:
+          command: 
+```
+
+Fill in the names for the Job that you're going to create, and the first Step that will run the Blender
+rendering script:
+
+```yaml
+specificationVersion: jobtemplate-2023-09
+name: DemoJob
+steps:
+  - name: BlenderRender
+    script:
+      actions:
+        onRun:
+          command: 
+```
+
+The way to think about a Step in Open Job Description is that it is defining a command to run and the collection of inputs
+to run the command with. This is a [map operation](https://en.wikipedia.org/wiki/Map_(higher-order_function)), if
+you are familiar with that concept. You're not going to define any inputs for the command yet, so you'll be using the default
+of having a single Task with empty input. The command is defined in the `onRun` Action in the skeleton above; this is
+the command that is run when running a Task within a [Session](How-Jobs-Are-Run#sessions). An Action in the 2023-09 
+revision of the specification [is defined as](2023-09-Template-Schemas#5-action):
+
+```yaml
+command: <CommandString> # @fmtstring[host]
+args: [ <ArgString>, ... ] # @optional @fmtstring[host]
+timeout: <posinteger> # @optional
+cancelation: <CancelationMethod> # @optional
+```
+
+Notice that both the `command` and elements of the `args` array are annotated with `@fmtstring[host]` indicating that
+they are [Format Strings](How-Jobs-Are-Constructed#format-strings) that are evaluated on the host that the command will
+run on. We're going to make use of this property to create our Job.
+
+You've already expressed your Job as a series of two shell scripts to run in sequence, so next you'll make use of the
```

Comment:
Feels like we skipped a step- we haven't had the OpenJD example of using the script itself yet, so "you've already expressed your job as.." doesn't make much sense. We've only been running single scripts, not any Jobs yet